tests/Aspire.Workload.Tests/WorkloadTestsBase.cs

```diff
@@ -0,0 +1,252 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Playwright;
+using Xunit;
+using Xunit.Abstractions;
+using static Aspire.Workload.Tests.TestExtensions;
+
+namespace Aspire.Workload.Tests;
+
+public class WorkloadTestsBase
+{
+    private static Lazy<IBrowser> Browser => new(CreateBrowser);
+    protected readonly TestOutputWrapper _testOutput;
+
+    public WorkloadTestsBase(ITestOutputHelper testOutput)
+        => _testOutput = new TestOutputWrapper(testOutput);
+
+    private static IBrowser CreateBrowser()
+    {
+        var t = Task.Run(async () =>
+        {
+            var playwright = await Playwright.CreateAsync();
+            BrowserTypeLaunchOptions? options = null;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                options = new BrowserTypeLaunchOptions
+                {
+                    Headless = false,
+                    ExecutablePath = "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
```

Comment:
NIT: should this be configurable (via env var or something) or is it always guaranteed to be there?